analyzer/topology_server.go

```diff
@@ -275,30 +280,31 @@ func (t *TopologyServer) OnWSJSONMessage(c shttp.WSSpeaker, msg *shttp.WSJSONMes
 	// this kind of message usually comes from external clients like the WebUI
 	if msgType == graph.SyncRequestMsgType {
 		t.Graph.RLock()
+		defer t.Graph.RUnlock()
+		delete(t.subscribers, c.GetHost())
+
 		syncMsg, status := obj.(graph.SyncRequestMsg), http.StatusOK
 		g, err := t.Graph.WithContext(syncMsg.GraphContext)
 		var result interface{} = g
+
 		if err != nil {
 			logging.GetLogger().Errorf("analyzer is unable to get a graph with context %+v: %s", syncMsg, err.Error())
-			result, status = nil, http.StatusBadRequest
+			t.sendSyncReply(c, msg, nil, status)
+			return
 		}
-
 		if syncMsg.GremlinFilter != "" {
 			subscriber, err := t.newTopologySubscriber(c.GetHost(), syncMsg.GremlinFilter)
 			if err != nil {
 				logging.GetLogger().Error(err)
+				t.sendSyncReply(c, msg, result, status)
```

Comment:
same as above

Reply:
my idea was that if user inserts incorrect query, he will receive the whole graph, but ur solution makes sense so I change it.